api/v1alpha1/corsproxy_types.go

```diff
@@ -0,0 +1,178 @@
+	}
+
+	corsproxyDefaultGrafanaDashboard defaultGrafanaDashboardSpec = defaultGrafanaDashboardSpec{
+		SelectorKey:   pointer.StringPtr("monitoring-key"),
+		SelectorValue: pointer.StringPtr("middleware"),
+	}
+)
+
+// CORSProxySpec defines the desired state of CORSProxy
+type CORSProxySpec struct {
+	// Image specification for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Image *ImageSpec `json:"image,omitempty"`
+	// Pod Disruption Budget for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	PDB *PodDisruptionBudgetSpec `json:"pdb,omitempty"`
+	// Horizontal Pod Autoscaler for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	HPA *HorizontalPodAutoscalerSpec `json:"hpa,omitempty"`
+	// Number of replicas (ignored if hpa is enabled) for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+	// Resource requirements for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Resources *ResourceRequirementsSpec `json:"resources,omitempty"`
+	// Liveness probe for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	LivenessProbe *ProbeSpec `json:"livenessProbe,omitempty"`
+	// Readiness probe for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	ReadinessProbe *ProbeSpec `json:"readinessProbe,omitempty"`
+	// Configures the Grafana Dashboard for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	GrafanaDashboard *GrafanaDashboardSpec `json:"grafanaDashboard,omitempty"`
+	// Application specific configuration options for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	Config CORSProxyConfig `json:"config"`
+}
+
+// Default implements defaulting for the CORSProxy resource
+func (a *CORSProxy) Default() {
+
+	a.Spec.Image = InitializeImageSpec(a.Spec.Image, corsproxyDefaultImage)
+	a.Spec.HPA = InitializeHorizontalPodAutoscalerSpec(a.Spec.HPA, corsproxyDefaultHPA)
+
+	if a.Spec.HPA.IsDeactivated() {
+		a.Spec.Replicas = intOrDefault(a.Spec.Replicas, &corsproxyDefaultReplicas)
+	} else {
+		a.Spec.Replicas = nil
+	}
+
+	a.Spec.PDB = InitializePodDisruptionBudgetSpec(a.Spec.PDB, corsproxyDefaultPDB)
+	a.Spec.Resources = InitializeResourceRequirementsSpec(a.Spec.Resources, corsproxyDefaultResources)
+	a.Spec.LivenessProbe = InitializeProbeSpec(a.Spec.LivenessProbe, corsproxyDefaultProbe)
+	a.Spec.ReadinessProbe = InitializeProbeSpec(a.Spec.ReadinessProbe, corsproxyDefaultProbe)
+	a.Spec.GrafanaDashboard = InitializeGrafanaDashboardSpec(a.Spec.GrafanaDashboard, corsproxyDefaultGrafanaDashboard)
+	a.Spec.Config.Default()
+}
+
+// CORSProxyConfig defines configuration options for the component
+type CORSProxyConfig struct {
+	// ets the nginx log level
```

Comment:
Wrong var decription, nothing to see with nginx log level

Reply:
Fixed